src/script/lua_api/l_util.cpp

```diff
@@ -398,7 +398,8 @@ int ModApiUtil::l_get_dir_list(lua_State *L)
 {
 	NO_MAP_LOCK_REQUIRED;
 	const char *path = luaL_checkstring(L, 1);
-	short is_dir = lua_isboolean(L, 2) ? lua_toboolean(L, 2) : -1;
+	bool list_all = lua_type(L, 2) != LUA_TBOOLEAN; // if its not a boolean list all
```

Comment:
not sure if i even can change this without causing even more trouble with that sightly broken github stuff. My plan was to delete my git copy and clone it again after this patch is not longer required. 

Like this guy: 😛 
![](http://imgs.xkcd.com/comics/git.png)